### PR TITLE
Fix explicit generation of static conversion methods in interfaces

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/61263

### CODE REVIEW

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/CodeGeneration/ConversionGenerator.cs [hunk 1]
```diff
@@ -100,11 +98,19 @@ private static ConversionOperatorDeclarationSyntax UseExpressionBodyIfDesired(
...
+        // Only "static" allowed if we're an explicit impl.
+        if (method.ExplicitInterfaceImplementations.Any())
+        {
+            return method.IsStatic ? [StaticKeyword] : [];
+        }
+        else
+        {
+            // If these appear in interfaces they must be static abstract
+            return destination is CodeGenerationDestination.InterfaceType
+                ? [StaticKeyword, AbstractKeyword]
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/CodeGeneration/ConversionGenerator.cs:112]:
```suggestion
                ? ([StaticKeyword, AbstractKeyword])
```

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/CodeGeneration/ConversionGenerator.cs [hunk 2]
```diff
@@ -100,11 +98,19 @@ private static ConversionOperatorDeclarationSyntax UseExpressionBodyIfDesired(
...
+                : [PublicKeyword, StaticKeyword];
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/CodeGeneration/ConversionGenerator.cs:113]:
```suggestion
                : ([PublicKeyword, StaticKeyword]);
```

**Comment by ToddGrun** [comment]
looks good, afaict

**Review by ToddGrun — APPROVED** [review]
:shipit:

**Pull request merged**